app/ts/ManualInput/ManualInputTable/CriteriaRows/CriterionRow/DataSourceRow/ReferenceCell/ReferenceCell.tsx

```diff
@@ -15,25 +16,62 @@ export default function ReferenceCell({
   criterion: ICriterion;
 }) {
   const {setDataSource} = useContext(ManualInputContext);
+  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
 
-  function handleChange(newReference: string) {
-    setDataSource(criterion.id, {...dataSource, reference: newReference});
+  const tooltipText = 'Edit reference';
+
+  function updateDataSource(newReference: string, newReferenceLink: string) {
+    setDataSource(criterion.id, {
+      ...dataSource,
+      reference: newReference,
+      referenceLink: newReferenceLink
+    });
+    closeDialog();
+  }
+
+  function renderReference(): JSX.Element {
+    if (
+      dataSource.reference &&
+      dataSource.referenceLink &&
+      !checkIfLinkIsInvalid(dataSource.referenceLink)
+    ) {
+      return (
+        <Tooltip title={tooltipText}>
+          <>
+            <div>{dataSource.reference}</div>
+            <div>({dataSource.referenceLink})</div>
```

Comment:
why not show the referencelink if it's invalid? it seems more informative to show it but maybe do markup if it's invalid?

Reply:
Can't enter invalid link, i.e. can't commit the changes in reference dialog